lib/rules/no-controller-access-in-routes.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+const ember = require('../utils/ember');
+const types = require('../utils/types');
+const { getImportIdentifier } = require('../utils/import');
+
+const ERROR_MESSAGE =
+  'Do not access controller in route outside of setupController/resetController';
+
+//------------------------------------------------------------------------------
+// Routing - No controller access in routes
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'disallow routes from accessing the controller outside of setupController/resetController',
+      category: 'Routes',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-controller-access-in-routes.md',
+    },
+    fixable: null,
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allowControllerFor: {
+            type: 'boolean',
+            default: false,
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+
+  ERROR_MESSAGE,
+
+  create(context) {
+    const allowControllerFor = context.options[0] && context.options[0].allowControllerFor;
+    let importedGetName;
+    let importedGetPropertiesName;
+    let inEmberRoute = false;
+
+    return {
+      ImportDeclaration(node) {
+        if (node.source.value === '@ember/object') {
+          importedGetName = importedGetName || getImportIdentifier(node, '@ember/object', 'get');
+          importedGetPropertiesName =
+            importedGetPropertiesName ||
+            getImportIdentifier(node, '@ember/object', 'getProperties');
+        }
+      },
+
+      ClassDeclaration(node) {
+        inEmberRoute = ember.isEmberRoute(context, node);
+      },
+      'ClassDeclaration:exit'() {
+        inEmberRoute = false;
+      },
+
+      MemberExpression(node) {
+        if (
+          inEmberRoute &&
+          types.isThisExpression(node.object) &&
+          node.property.name === 'controller'
```

Comment:
Should add this check before checking the property name: `types.isIdentifier(node.property)`

Reply:
Good catch, added.